packages/cdktf-cli/lib/checkpoint.ts

```diff
@@ -94,6 +102,20 @@ export async function ReportRequest(reportParams: ReportParams): Promise<void> {
     reportParams.os = os.platform();
   }
 
+  const ci: string | false = ciDetect();
+  if (!reportParams.userId && !ci) {
+    reportParams.userId = machineIdSync();
+  }
+
+  if (ci) {
+    reportParams.ci = ci;
+  }
+
+  const gitRemote = await gitRemoteOriginUrl();
```

Comment:
This will raise an error when it's not a git repo - see https://github.com/sindresorhus/git-remote-origin-url/blob/8573705e8f0f482708f0a63413cecfdde4e31d19/index.js#L12

Reply:
Also, isn't this something like a project fingerprint we could create once as a uuid on project init and add it to `cdktf.json`?

Reply:
That would be a different solution, but yeah we can add a project fingerprint to the cdktf.json if this seems like a better solution to you

Reply:
Not sure if it's better. It's harder to retrofit the uuid approach for existing projects and we'd have to find a way to pass the value through to this point. The upside is, that we wouldn't have to re-run this for each and every command and that it's not depending on git.

Reply:
Downside is that if you copy and paste instead of using init we'll get wrong data. I'll add a try catch and we'll see if this approach works well enough